lib/consumer.js

```diff
@@ -199,16 +156,7 @@ Consumer.prototype.getSchema = function(index) {
  */
 Consumer.prototype.getSchemas = function() {
   var self = this;
-  return new Promise(function(fulfill, reject) {
-    self.registry.getSchemasLenght.call({from: self.address}, function(error, length) {
-      if (error) {
-        console.error(error);
-        reject(error);
-        return;
-      }
-      fulfill(length);
-    });
-  }).then(function(length) {
+  return self.registry.getSchemasLenghtAsync({from: self.address}).then(function(length) {
```

Comment:
Same style comments with using `self` and arrow functions and `then` block as before. Also, the `getSchemasLenghtAsync` method has length misspelled. Is this something we control or is this in the library?